cmd/lifecycle/exporter.go

```diff
@@ -153,15 +163,27 @@ func (e *exportCmd) Exec() error {
 		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "parse analyzed metadata")
 	}
 
-	cacheStore, err := initCache(e.cacheImageTag, e.cacheDir)
+	cacheStore, err := initCache(e.cacheImageTag, e.cacheDir, e.keychain)
 	if err != nil {
 		cmd.DefaultLogger.Infof("no stack metadata found at path '%s', stack metadata will not be exported\n", e.stackPath)
 	}
 
-	return e.export(group, cacheStore, analyzedMD)
+	return e.export(group, cacheStore, analyzedMD, e.keychain)
+}
+
+func (e *exportCmd) resolveKeychain() (authn.Keychain, error) {
+	var registryImages []string
+	if e.cacheImageTag != "" {
+		registryImages = append(registryImages, e.cacheImageTag)
+	}
+	if !e.useDaemon {
+		registryImages = append(registryImages, e.imageNames...)
+		registryImages = append(registryImages, e.runImageRef)
```

Comment:
I think we have a small problem in that, at this point, e.runImageRef might be empty and the best run image is not located until https://github.com/buildpacks/lifecycle/pull/409/files#diff-92c6ed289675df1e2edf598694dfc99f22a591496c130d10a43e86830cbd3686R193

We may need to reorganize the run image logic a bit.

Reply:
Moved this resolution up into `Args()`